packages/react/src/core/utils/useFocus.test.tsx

```diff
@@ -0,0 +1,79 @@
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import { useRef } from 'react';
+import { render, screen } from '../../../../../test-utils';
+import { UseFocusParams, useFocus } from './useFocus';
+
+const ComponentWithUseFocus = (
+	props: Partial<UseFocusParams<HTMLDivElement>>
+) => {
+	const ref = useFocus(props);
+
+	return (
+		<>
+			<div data-testid="div-with-ref" ref={ref} tabIndex={-1} />
+			<div data-testid="div-with-no-ref" tabIndex={-1} />
+		</>
+	);
+};
+
+describe('Given useFocus is called', () => {
+	describe('when no parameters are defined', () => {
+		test('then the attached component should be focused once on mount', () => {
+			render(<ComponentWithUseFocus />);
+			expect(screen.getByTestId('div-with-ref')).toHaveFocus();
+		});
+	});
+
+	describe('when focusOnUpdate is defined', () => {
+		test('then the attached component should be focused when the value is updated', async () => {
+			const { rerender } = render(
+				<ComponentWithUseFocus focusOnUpdate={['mock-value-1']} />
+			);
+
+			const user = userEvent.setup();
+			await user.tab();
+
+			rerender(<ComponentWithUseFocus focusOnUpdate={['mock-value-2']} />);
```

Comment:
This is interesting, because it doesn't test how a consumer will use this. The docs state: "It'd probably be better if you test the component that's doing the prop updating to ensure that the props are being updated correctly", which I kinda agree with. This test seems confusing at first glance

Reply:
Sorry, I don't follow. Which docs are these?

Reply:
Sorry, the testing library docs

Reply:
Oh right, I'm with you. So, the reason I'm using `rerender` on the test component (`ComponentWithUseFocus`) is because it's not the target of the test. Exactly how the test component updates props or what the value of those props are doesn't matter. The target is the `useFocus` hook and its effect on the attached element. So, we pass the change in value to `updateOnFocus` via the test component and find the attached element on the `screen` to test the effect on the attached element.

I can add comments to explain what's happening, if that's helpful?

- [ ] annotate with comments

Reply:
Yeah, some comments to that effect may help 🙏